### ISSUE TITLE
peer dependencies issue

### ISSUE THREAD

**Issue by petrakoww** [SoftwareBrothers/adminjs#1196]
**Describe the bug**
While installing version 6.0.3 of adminjs, I get a bunch of warnings for unresolved dependencies which are resolved by overriding peer deps:

```
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-redux@7.2.0
npm WARN Found: react@18.2.0
npm WARN node_modules/adminjs/node_modules/react
npm WARN   react@"^18.1.0" from adminjs@6.0.3
npm WARN   node_modules/adminjs
npm WARN     adminjs@"^6.0.3" from the root project
npm WARN     2 more (@adminjs/express, @adminjs/mongoose)
npm WARN   2 more (@adminjs/design-system, react-dom)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.3" from react-redux@7.2.0
npm WARN node_modules/adminjs/node_modules/react-redux
npm WARN   react-redux@"=7.2.0" from adminjs@6.0.3
npm WARN   node_modules/adminjs
npm WARN 
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN node_modules/react
npm WARN   peer react@"^16.8.3" from react-redux@7.2.0
npm WARN   node_modules/adminjs/node_modules/react-redux
npm WARN     react-redux@"=7.2.0" from adminjs@6.0.3
npm WARN     node_modules/adminjs
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-text-mask@5.4.3
npm WARN Found: react@18.2.0
npm WARN node_modules/adminjs/node_modules/react
npm WARN   react@"^18.1.0" from adminjs@6.0.3
npm WARN   node_modules/adminjs
npm WARN     2 more (@adminjs/express, @adminjs/mongoose)
npm WARN   2 more (@adminjs/design-system, react-dom)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14.0 || ^15.0.0 || ^16.0.0" from react-text-mask@5.4.3
npm WARN node_modules/adminjs/node_modules/react-text-mask
npm WARN   react-text-mask@"^5.4.3" from @adminjs/design-system@3.0.0
npm WARN   node_modules/adminjs/node_modules/@adminjs/design-system
npm WARN
npm WARN Conflicting peer dependency: react@16.14.0
npm WARN   peer react@"^0.14.0 || ^15.0.0 || ^16.0.0" from react-text-mask@5.4.3
npm WARN   node_modules/adminjs/node_modules/react-text-mask
npm WARN     react-text-mask@"^5.4.3" from @adminjs/design-system@3.0.0
npm WARN     node_modules/adminjs/node_modules/@adminjs/design-system
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated @babel/polyfill@7.12.1: 🚨 This package has been deprecated in favor of separate inclusion of a polyfill and regenerator-runtime (wh
en needed). See the @babel/polyfill docs (https://babeljs.io/docs/en/babel-polyfill) for more information.
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8
 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compati
bility issues. Please, upgrade your dependencies to the actual version of core-js.

```

**Installed libraries and their versions**

* adminjs@6.0.3
* @adminjs/express@5.0.0
* @adminjs/mongoose@3.0.0
* express@4.18.1
* mongodb@4.8.0
* mongoose@6.4.4
* node version 18.5.0
* npm version 8.12.1

**To Reproduce**
1. npm i --save adminjs6.0.3 @adminjs/express@5.0.0 @adminjs/mongoose@3.0.0

**Expected behavior**
Successful npm installation.

**Desktop (please complete the following information if relevant):**
 - OS: Windows 10
 - Browser: Chrome

**Additional context**
I think you should update the versions of the peer dependencies,  so they are matching the dependencies.

**Comment by petrakoww** [comment]
If i use npm install -f

**Comment by dziraf** [comment]
> If i use npm install -f i get this error:
> 
> ```
>  throw Error( "Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the followin
> g reasons:\n1. You might have mismatching versions of React and the renderer (such as React DOM)\n2. You might be breaking the Rules of Hooks\n3. You
>  might have more than one copy of React in the same app\nSee https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this prob
> lem." );
>             ^
> 
> Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:   
> 1. You might have mismatching versions of React and the renderer (such as React DOM)
> 2. You might be breaking the Rules of Hooks
> 3. You might have more than one copy of React in the same app
> See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
>     at resolveDispatcher (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\react\cjs\react.development.js:1476:13)
>     at Object.useState (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\react\cjs\react.development.js:1507:20)
>     at ae (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\styled-components\dist\styled-components.cjs.js:1:13285)
>     at renderWithHooks (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-server-legacy.no
> de.development.js:5662:16)
>     at renderIndeterminateComponent (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-ser
> ver-legacy.node.development.js:5735:15)
>     at renderElement (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-server-legacy.node
> .development.js:5950:7)
>     at renderNodeDestructiveImpl (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-server
> -legacy.node.development.js:6108:11)
>     at renderNodeDestructive (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-server-leg
> acy.node.development.js:6080:14)
>     at retryTask (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-server-legacy.node.dev
> elopment.js:6532:5)
>     at performWork (C:\Users\vasil\Desktop\app\d3d-main-api\node_modules\adminjs\node_modules\react-dom\cjs\react-dom-server-legacy.node.d
> evelopment.js:6580:7)
> 
> Node.js v18.5.0
> [nodemon] app crashed - waiting for file changes before starting...
> ```
Please see if this comment helps you:
https://github.com/SoftwareBrothers/adminjs/issues/1180#issuecomment-1190293422

I'll have a look at peer deps issue today/tomorrow

**Comment by krzysztofstudniarek** [comment]
It's a duplicate of https://github.com/SoftwareBrothers/adminjs/issues/1180#issuecomment-1190293422
